### PR TITLE
main/py-sqlalchemy: upgrade to 1.3.1, move back to community

### PR DESCRIPTION
**Pull request by dimon222**
https://docs.sqlalchemy.org/en/latest/changelog/changelog_13.html#change-1.3.1

### CODE REVIEW

**Comment by andypost** [comment]
Meantime 1.2.16 https://docs.sqlalchemy.org/en/latest/changelog/changelog_12.html#change-1.2.16

**Comment by dimon222** [comment]
Good point, i'll bump and check it.
I also noticed that it has test in their repo too. I guess nothing wrong with adding them to this build?

**Comment by andypost** [comment]
Btw tests show that not all dependencies installed, it means this packages should be updated
```
running test
Searching for pytest-xdist
Reading https://pypi.org/simple/pytest-xdist/
Downloading https://files.pythonhosted.org/packages/72/e6/f07fbf71cd2cc8c5160ef2fb5ce61591dcfe39a2ff747b1add29580553f2/pytest_xdist-1.26.0-py2.py3-none-any.whl#sha256=5795f665e112520fa5beab736ad957e7f36ce7d44210f4004be9d99f86529d97
Best match: pytest-xdist 1.26.0
Processing pytest_xdist-1.26.0-py2.py3-none-any.whl
Installing pytest_xdist-1.26.0-py2.py3-none-any.whl to /mnt/community/py-sqlalchemy/src/SQLAlchemy-1.2.16/.eggs
writing requirements to /mnt/community/py-sqlalchemy/src/SQLAlchemy-1.2.16/.eggs/pytest_xdist-1.26.0-py2.7.egg/EGG-INFO/requires.txt

Installed /mnt/community/py-sqlalchemy/src/SQLAlchemy-1.2.16/.eggs/pytest_xdist-1.26.0-py2.7.egg
Searching for pytest-forked
Reading https://pypi.org/simple/pytest-forked/
Downloading https://files.pythonhosted.org/packages/b0/2a/99ec2a8a26c0a911f76ec2a9461ff0e0d97b5b7bd3282de57eca947f39fc/pytest_forked-1.0.1-py2.py3-none-any.whl#sha256=8d05c2e6f33cd4422571b2b1bb309720c398b0549cff499e3e4cde661875ab54
Best match: pytest-forked 1.0.1
Processing pytest_forked-1.0.1-py2.py3-none-any.whl
Installing pytest_forked-1.0.1-py2.py3-none-any.whl to /mnt/community/py-sqlalchemy/src/SQLAlchemy-1.2.16/.eggs
writing requirements to /mnt/community/py-sqlalchemy/src/SQLAlchemy-1.2.16/.eggs/pytest_forked-1.0.1-py2.7.egg/EGG-INFO/requires.txt

Installed /mnt/community/py-sqlalchemy/src/SQLAlchemy-1.2.16/.eggs/pytest_forked-1.0.1-py2.7.egg
Searching for execnet>=1.1
Reading https://pypi.org/simple/execnet/
Downloading https://files.pythonhosted.org/packages/f9/76/3343e69a2a1602052f587898934e5fea395d22310d39c07955596597227c/execnet-1.5.0-py2.py3-none-any.whl#sha256=fc155a6b553c66c838d1a22dba1dc9f5f505c43285a878c6f74a79c024750b83
Best match: execnet 1.5.0
Processing execnet-1.5.0-py2.py3-none-any.whl
Installing execnet-1.5.0-py2.py3-none-any.whl to /mnt/community/py-sqlalchemy/src/SQLAlchemy-1.2.16/.eggs
writing requirements to /mnt/community/py-sqlalchemy/src/SQLAlchemy-1.2.16/.eggs/execnet-1.5.0-py2.7.egg/EGG-INFO/requires.txt

Installed /mnt/community/py-sqlalchemy/src/SQLAlchemy-1.2.16/.eggs/execnet-1.5.0-py2.7.egg
Searching for apipkg>=1.4
Reading https://pypi.org/simple/apipkg/
Downloading https://files.pythonhosted.org/packages/67/08/4815a09603fc800209431bec5b8bd2acf2f95abdfb558a44a42507fb94da/apipkg-1.5-py2.py3-none-any.whl#sha256=58587dd4dc3daefad0487f6d9ae32b4542b185e1c36db6993290e7c41ca2b47c
Best match: apipkg 1.5
Processing apipkg-1.5-py2.py3-none-any.whl
Installing apipkg-1.5-py2.py3-none-any.whl to /mnt/community/py-sqlalchemy/src/SQLAlchemy-1.2.16/.eggs

Installed /mnt/community/py-sqlalchemy/src/SQLAlchemy-1.2.16/.eggs/apipkg-1.5-py2.7.egg
running egg_info
writing requirements to lib/SQLAlchemy.egg-info/requires.txt
```
PS first commit no longer needed `upgrade to 1.2.15`

**Comment by dimon222** [comment]
I'll rebase to remove redundant upgrade. I didn't find those dependencies in alpine repo tho, so its in comments
https://github.com/alpinelinux/aports/pull/6030/files#diff-4bebe7ecda9f1391f5ba718bd4fd8a20R14

**Comment by andypost** [comment]
Would be great to get @kaniini feedback on it

**Comment by andypost** [comment]
First PR commited, next https://github.com/alpinelinux/aports/pull/6053

**Comment by dimon222** [comment]
Bumped version to 1.2.17, applied rebase into single commit, cleaned dependencies a bit.
Still requires pytest-xdist (#6090) to run tests

**Comment by andypost** [comment]
@kaniini all dependencies commited to testing, we should move them to community.
@dimon222 which packages except `testing/pytest-xdist` should be moved?

**Comment by dimon222** [comment]
@andypost 

tl;dr just the nested reqs of `testing/pytest-xdist`
For runtime - `testing/py-execnet`, `testing/pytest-forked`, `testing/py-apipkg`
For build - `testing/py-setuptools_scm`
For test/check - `testing/py-filelock`

**Comment by dimon222** [comment]
Not sure why the build/tests fail, pytest-xdist is already built. Do we need to retrigger build?

**Comment by andypost** [comment]
I did restart travis

**Comment by dimon222** [comment]
Bumped version to 1.3.1. Now dependencies part of main, hopefully it will resolve the build problem.

**Comment by andypost** [comment]
It needs rebase after edf2ae4e43

**Comment by andypost** [comment]
Package moved to main, also `testing/py-sqlalchemy7` which is duplicate

**Comment by dimon222** [comment]
I applied rebase (seems somebody already added tests, eh) with just 1.3.1 upgrade and added missing pytest-xdist dependency.

Build is failing, for some reason it cant find community/pytest-xdist dependency

**Comment by dimon222** [comment]
Due to non-usage of pytest-xdist tests are taking forever (not parallelized). :(
I'm looking if I can add usage of it into this PR

**Comment by andypost** [comment]
Looks great to go! Needs someone with permissions to move

**Comment by clandmeter** [comment]
merged thx!
